Name: Bookmarks-List Button In Browser
Version: 1.3.5-1
Author: Jason Robitaille
Description: Adds a bookmarks-list button that brings up the list view of bookmarks  (MIT license)

--- .orig/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
@@ -10,6 +10,12 @@
 		command: 'back'
 	},
 	
+	Bookmarks: {
+		label: $L('Bookmarks'),
+		icon: 'bookmarks',
+		command: 'show-bookmarks-cmd'
+	},
+	
 	Forward: {
 		label: $L('Go forward'),
 		icon: 'forward',
--- .orig/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
@@ -25,6 +25,13 @@
 			icon: MenuData.NavigationMenu.Forward.icon,
 			command: MenuData.NavigationMenu.Forward.command
 		};
+		
+		this._bookmarksModel = {
+			disabled: false,
+			label: MenuData.NavigationMenu.Bookmarks.label,
+			icon: MenuData.NavigationMenu.Bookmarks.icon,
+			command: MenuData.NavigationMenu.Bookmarks.command
+		};
 
 		this._model = {
 			visible: true,
@@ -28,7 +35,7 @@
 
 		this._model = {
 			visible: true,
-			items: [this._backModel, {}, {}]
+			items: [this._backModel, this._bookmarksModel, {}, {}]
 		};
 
 		this._currentPageModel = {};
@@ -64,6 +71,7 @@
 		var items = $A();
 		this._backModel.disabled = !back;
 		items.push(this._backModel);
+		items.push(this._bookmarksModel);
 
 		if (forward) {
 			this._forwardModel.disabled = !forward;
